File: docs-examples/example-java/src/test/java/io/micronaut/mqtt/docs/custom/annotation/ProductClient.java
package io.micronaut.mqtt.docs.custom.annotation;

import io.micronaut.mqtt.annotation.Topic;
import io.micronaut.mqtt.v5.annotation.MqttPublisher;
import io.micronaut.context.annotation.Requires;

@Requires(property = "spec.name", value = "CorrelationSpec")
@MqttPublisher
public interface ProductClient {

    @Topic("product")
    void send(byte[] data, @Correlation byte[] correlation);
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/mqtt/docs/custom/annotation/ProductClient.kt
package io.micronaut.mqtt.docs.custom.annotation

import io.micronaut.mqtt.v5.annotation.MqttPublisher
import io.micronaut.context.annotation.Requires
import io.micronaut.mqtt.annotation.Topic

@Requires(property = "spec.name", value = "CorrelationSpec")
@MqttPublisher
interface ProductClient {

    @Topic("product")
    fun send(@Correlation data: ByteArray)
}
